src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -420,4 +420,14 @@ public function testVeryLongQuotedStrings()
 
         $this->assertEquals($longStringWithQuotes, $arrayFromYaml['longStringWithQuotes']);
     }
+
+    public function testBooleanMappingKeysAreConvertedToStrings()
+    {
+        $this->assertSame(array('false' => 'foo'), Inline::parse('{false: foo}'));
```

Comment:
see http://www.yaml.org/spec/1.2/spec.html#id2802432:

> YAML places no restrictions on the type of keys; in particular, they are not restricted to being scalars.

Keys can be absolutely anything in YAML, including another mapping.